java8/src/main/java/Client/LogicalOperations.java

```diff
@@ -0,0 +1,184 @@
+                        .max(Comparator.comparing(s -> s.valueOf(s))));
+        System.out.println(max);
+    }
+
+    static int countElementsOver10(Map<String, Integer> map) {
+
+        int count = 0;
+        for (Map.Entry<String, Integer> entry : map.entrySet()) {
+            Integer v = entry.getValue();
+            if (v > 10) count++;
+        }
+        return count;
+    }
+
+    static void printIfEligible(List<Car> carsList) {
+        carsList.stream().filter(LogicalOperations::isEligible).map(car -> car.getModel().toUpperCase() + " has 4 doors.").forEach(System.out::println);
+    }
+
+    private static boolean isEligible(Car car) {
+        if (car.getDoors() != 4) {
+            System.out.println(car.getModel().toUpperCase() + " is NOT eligible because it has  " + car.getDoors() + " doors, but let's print it anyways.");
+            return false;
+        }
+        return true;
+    }
+
+    static void sortObjectsByHashCode(List<Object> objectList) {
+        List<Object> sortedList = objectList.stream()
+                .sorted(Comparator.comparing(Object::hashCode))
+                .collect(Collectors.toList());
+        System.out.println(sortedList);
+    }
+
+    static void compareReversed(HashSet<String> set) {
+
+        TreeSet<String> collection = set.stream()
+                .collect(Collectors.toCollection(TreeSet::new));
+        TreeSet<String> reversed = (TreeSet<String>) collection.descendingSet();
+        System.out.println(reversed);
+    }
+
+    static void findMinByName(HashSet<Car> set) {
+        Optional<Car> min = set.stream()
+                .min(Comparator.comparing(car -> car.getModel().length()));
+
+        System.out.println(min);
+    }
+
+    private static int generateRandomInt() {
+        int min = 1;
+        int max = 15;
+        Random random = new Random();
+        return random.nextInt((max - min) + 1 + min);
+    }
+
+    static List<Integer> throwDice() {
+        List<Integer> genList = new ArrayList<>();
+        int min = 1;
+        int max = 5;
+        Random r = new Random();
+        int factor = r.nextInt((max - min) + 1 + min) + 1;
+        for (int i = 0; i < factor; i++) {
+            int generated = generateRandomInt();
+            genList.add(generated);
+        }
+        return genList;
+    }
+
+    static Map<Integer, Integer> createExponentialsMap(List<Integer> integerList) {
+        Map<Integer, Integer> intMap = new LinkedHashMap<>();
+        for (Integer integer : integerList) {
+            int exponential = integer * integer;
+            intMap.put(integer, exponential);
+        }
+        return intMap;
+    }
+
+    public static int countInts(List<Integer> intList) {
+        long count = intList.stream()
+                .count();
+        return (int) count;
+    }
+
+    static void optionalWrappingExample() {
+        //optional
+        List<Car> carList = new ArrayList<>();
+        Car c1 = new Car("Fiat", true, 3);

```

Comment:
naming, re-check them everywhere (suggestion for you) try focusing on them as well)